src/Command/Ingester/Native/NativeUserIngesterCommand.php

```diff
@@ -178,58 +185,113 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $progressBar->setMaxSteps($numberOfUsers);
             $progressBar->start();
 
-            $index = $this->getAvailableStartIndex();
             $lineItemCollection = $this->fetchLineItems();
 
-            foreach ($source->getContent() as $row) {
+            $batchedRawUsers = [];
+            $numberOfProcessedRows = 1;
+            foreach ($source->getContent() as $rawUser) {
+                $batchedRawUsers[] = $rawUser;
+                if ($numberOfProcessedRows % $this->batchSize === 0) {
+                    $this->processBatchedRawUsers($batchedRawUsers, $lineItemCollection);
+                    $batchedRawUsers = [];
+                    $progressBar->advance($this->batchSize);
+                }
+
+                $numberOfProcessedRows++;
+            }
+
+            if ($batchedRawUsers) {
+                $this->processBatchedRawUsers($batchedRawUsers, $lineItemCollection);
+            }
+        } catch (Throwable $exception) {
+            $this->errors[] = $exception->getMessage();
+            $this->userQueryParts = [];
+            $this->assignmentQueryParts = [];
+        } finally {
+            $this->refreshSequences($resultSetMapping);
+            $progressBar->finish();
+        }
+
+        foreach ($this->errors as $error) {
+            $this->symfonyStyle->error($error);
+        }
+
+        return (int)!empty($this->errors);
```

Comment:
Are you sure that you want to return only 0 or 1?
in case of yes, we could return a boolean, no?

Reply:
In linux shell environment command return value (exit status) is traditionally zero in case of success, or nonzero integer in case of error (different exit status for every possible errors). This is also enforced by interface contract of Symfony Console component:

```
protected function execute(InputInterface $input, OutputInterface $output): int
```